### PR TITLE
feat: Authorization Modal

### LINKED ISSUE #1569: Create new authorization modal
## 🎉 Description

Create new authorization modal component with the possibility to add multiple steps:

The actions are not included in this card

* [ ] 🐛 This is a bug report.
* [ ] 📈 This is a feature request.

### PR DESCRIPTION
**Pull request by meelrossi**
Add authorization modal logic with different steps

https://user-images.githubusercontent.com/11800206/232871976-ce8c658d-44bb-4566-825f-be50ceb3ee14.mov

Closes #1569

### CODE REVIEW

File: webapp/src/components/BuyPage/BuyNFTModal/BuyNFTModal.tsx [hunk 1]
```diff
@@ -75,68 +73,39 @@ const BuyNFTModal = (props: Props) => {
...
-        )) ||
-      isBuyWithCardPage
-    ) {
+    if (isBuyWithCardPage) {
       handleExecuteOrder()
-    } else {
-      setShowAuthorizationModal(true)
+      return;
     }
+    onAuthorizedAction(authorization, order?.price || '', handleExecuteOrder)
```
Comment (juanmahidalgo) [inline webapp/src/components/BuyPage/BuyNFTModal/BuyNFTModal.tsx:100]:
```suggestion
    !!order?.price && onAuthorizedAction(authorization, order.price, handleExecuteOrder)
```
wdyt?

File: webapp/src/components/BuyPage/MintItemModal/MintItemModal.tsx [hunk 2]
```diff
@@ -65,68 +63,40 @@ const MintItemModal = (props: Props) => {
...
-      isBuyWithCardPage ||
-      +item.price === 0
-    ) {
+    if (isBuyWithCardPage) {
       handleExecuteOrder()
-    } else {
-      setShowAuthorizationModal(true)
+      return
     }
+    onAuthorizedAction(authorization, item?.price, handleExecuteOrder)
```
Comment (juanmahidalgo) [inline webapp/src/components/BuyPage/MintItemModal/MintItemModal.tsx:93]:
```suggestion
    !!item && onAuthorizedAction(authorization, item.price, handleExecuteOrder)
```

**Review by juanmahidalgo — COMMENTED** [review]
I like the approach taken by using the HOC! Some small suggestions, but besides that looks good :)

File: webapp/src/components/BuyPage/BuyNFTModal/BuyNFTModal.tsx (hunk 1 above)
Comment (meelrossi) [reply webapp/src/components/BuyPage/BuyNFTModal/BuyNFTModal.tsx:100]: not sure how it will handle price 0. is that a possible price value for an NFT?

**Review by juanmahidalgo — APPROVED** [review]
Looks great! Good job 🤟🏽

**Pull request merged**